### PR TITLE
Add limit query param to the API

### PR DESCRIPTION
**Pull request by angelbarrera92**
This PR enable passing a query parameter to the `group`s endpoints *(both plain groups and groups + target)*.
The query parameter is `limit`. The default `limit` is 100. 

Also changes the previous behaviour from returning every result to return the latest `limit` results.
Let me know if everything is fine :)

### CODE REVIEW

**Review by nandajavarma — APPROVED** [review]
Looks pretty good! I recommend to add the `limit` query string to the documentation.

File: docs/releases/v0.3.0.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+It exposes two simple endpoints:
+
+- `/group/:group`: Returns the jobs executed under a specific group
+- `/group/:group/target/:target`: Returns the jobs executed under a specific target and group
+
+Both returns a json structure with a base `results` attribute and the following object:
+
+```json
+{
+  "results":
```
Comment (nandajavarma) [inline docs/releases/v0.3.0.md:31]: @angelbarrera92 The json response currently do not have `results` key. It is basically just an indexed list of items iirc.

**Review by nandajavarma — COMMENTED** [review]
just a tiny query on the documentation

File: docs/releases/v0.3.0.md (hunk 1 above)
Comment (angelbarrera92) [reply docs/releases/v0.3.0.md:31]: 🤔 It has the results attribute

**Review by nandajavarma — APPROVED** [review]
LGTM!!

**Pull request merged**